src/main/java/br/com/apppersonal/apppersonal/service/PersonalService.java

```diff
@@ -0,0 +1,134 @@
+
+@Service
+public class PersonalService {
+    private UserRepository userRepository;
+
+    @Autowired
+    public PersonalService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public ResponseEntity<?> associateUserWithPersonal(Long userId, Long personalId) {
+        try {
+            if (userId == null) throw new ParameterNullException("Identificador do usuário não informado");
+            if (personalId == null) throw new ParameterNullException("Identificador do personal trainer não informado");
+
+            UserEntity students = userRepository.findByIdAndRole(userId, Role.USER);
+
+            if (students == null) {
+                throw new UserNotFoundException("Usuário não encontrado");
+            }
+
+            students.setPersonalTrainerId(personalId); // Associa o usuário ao personal trainer
+            userRepository.save(students);
+
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(
+                            new ApiResponse(
+                                    true,
+                                    "Usuário associado com sucesso ao personal trainer"
+                            )
+                    );
+        } catch (Exception e) {
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(
+                            new ApiResponse(
+                                    false,
+                                    e.getMessage()
+                            )
+                    );
+        }
+    }
+    public ResponseEntity<?> dissociateUserFromPersonal(Long userId) {
+        try {
+            if (userId == null) throw new ParameterNullException("Identificador do usuário não informado");
+
+            UserEntity students = userRepository.findByIdAndRole(userId, Role.USER);
+
+            if (students == null) {
+                throw new UserNotFoundException("Usuário não encontrado");
+            }
+
+            students.setPersonalTrainerId(null); // Remove a associação com o personal trainer
+            userRepository.save(students);
+
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(
+                            new ApiResponse(
+                                    true,
+                                    "Usuário dissociado com sucesso do personal trainer"
+                            )
+                    );
+        } catch (Exception e) {
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(
+                            new ApiResponse(
+                                    false,
+                                    e.getMessage()
+                            )
+                    );
+        }
+    }
+
+    public ResponseEntity<?> listStudentsByPersonal(Long personalId) {
```

Comment:
Adicionar comentário referente a função.

```suggestion
    /**
    * Lista alunos de um personal trainer
    *
    * @param  Long personalId
    * @return  ResponseEntity
    */
    public ResponseEntity<?> listStudentsByPersonal(Long personalId) {
```